File: backend/fudbalski_klub/src/main/java/com/football_club/Scouting/service/ScoutManagementService.java
package com.football_club.Scouting.service.impl;

import com.football_club.Auth.model.RoleEnum;
import com.football_club.Auth.model.User;
import com.football_club.Auth.repository.UserRepository;
import com.football_club.Scouting.dto.ScoutActivePlayerDTO;
import com.football_club.Scouting.dto.ScoutPerformanceDTO;
import com.football_club.Scouting.model.MonitoredPlayer;
import com.football_club.Scouting.repository.MonitoredPlayerRepository;
import com.football_club.Scouting.repository.ReportRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class ScoutManagementService {

    private final UserRepository userRepository;
    private final ReportRepository reportRepository;
    private final MonitoredPlayerRepository monitoredPlayerRepository;

    @Transactional(readOnly = true)
    public List<ScoutPerformanceDTO> getScoutPerformances() {
        List<User> scouts = userRepository.findByRole(RoleEnum.ROLE_SCOUT);

        return scouts.stream().map(scout -> ScoutPerformanceDTO.builder()
                .id(scout.getId())
                .name(scout.getName())
                .surname(scout.getSurname())
                .username(scout.getUsername())
                .email(scout.getEmail())
                .region(scout.getRegion())
                .totalReports(reportRepository.countByScoutId(scout.getId()))
                .activeMonitoredPlayers(monitoredPlayerRepository.countActiveByScoutId(scout.getId()))
                .build()
        ).collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public List<ScoutActivePlayerDTO> getActivePlayersForScout(Long scoutId) {
        List<MonitoredPlayer> activePlayers = monitoredPlayerRepository.findActiveByScoutId(scoutId);

        return activePlayers.stream().map(mp -> ScoutActivePlayerDTO.builder()
                .playerId(mp.getPlayer().getId())
                .playerName(mp.getPlayer().getName())
                .playerSurname(mp.getPlayer().getSurname())
                .photoUrl(mp.getPlayer().getPhotoUrl())
                .campaignName(mp.getCampaign().getName())
                .currentTeamName(mp.getPlayer().getCurrentTeam() != null ? mp.getPlayer().getCurrentTeam().getName() : "Slobodan igrač")
                .build()
        ).collect(Collectors.toList());
    }
}

File: frontend/football-club/src/app/feature-modules/scouting/services/scout-management.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from 'src/env/environment';

export interface ScoutPerformance {
  id: number;
  name: string;
  surname: string;
  username: string;
  email: string;
  region: string | null;
  totalReports: number;
  activeMonitoredPlayers: number;
}

export interface ScoutActivePlayer {
  playerId: number;
  playerName: string;
  playerSurname: string;
  photoUrl: string;
  campaignName: string;
  currentTeamName: string;
}

@Injectable({ providedIn: 'root' })
export class ScoutManagementService {
  private baseUrl = `${environment.apiHost}scout-management`;

  constructor(private http: HttpClient) {}

  getScoutPerformances(): Observable<ScoutPerformance[]> {
    return this.http.get<ScoutPerformance[]>(`${this.baseUrl}/performances`);
  }

  getActivePlayersForScout(scoutId: number): Observable<ScoutActivePlayer[]> {
    return this.http.get<ScoutActivePlayer[]>(`${this.baseUrl}/scouts/${scoutId}/active-players`);
  }

  updateScoutRegion(scoutId: number, region: string): Observable<string> {
    let params = new HttpParams().set('region', region);
    return this.http.patch(`${this.baseUrl}/scouts/${scoutId}/region`, null, { 
      params: params,
      responseType: 'text' 
    });
  }
}